File: necform/src/main/java/org/sid/necform/service/UtilisateurService.java
package org.sid.necform.service;

import org.sid.necform.dto.request.CreateUtilisateurRequest;
import org.sid.necform.dto.request.UpdateUtilisateurRequest;
import org.sid.necform.dto.response.UtilisateurResponse;

import java.util.List;
import java.util.UUID;

public interface UtilisateurService {

    UtilisateurResponse create(CreateUtilisateurRequest request);

    List<UtilisateurResponse> getAll();

    UtilisateurResponse getById(UUID id);

    UtilisateurResponse update(UUID id, UpdateUtilisateurRequest request);

    void delete(UUID id);
}

File: necform-front/src/app/core/services/utilisateur.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { 
  Utilisateur, 
  CreateUtilisateur, 
  UpdateUtilisateur 
} from '../../shared/models/utilisateur.model';
import { ApiService } from './api.service';

@Injectable({
  providedIn: 'root'
})
export class UtilisateurService {
  private readonly endpoint: string;

  constructor(
    private http: HttpClient,
    private apiService: ApiService
  ) {
    this.endpoint = `${this.apiService.getApiUrl()}/utilisateurs`;
  }

  create(request: CreateUtilisateur): Observable<Utilisateur> {
    return this.http.post<Utilisateur>(this.endpoint, request);
  }

  getAll(): Observable<Utilisateur[]> {
    return this.http.get<Utilisateur[]>(this.endpoint);
  }

  getById(id: string): Observable<Utilisateur> {
    return this.http.get<Utilisateur>(`${this.endpoint}/${id}`);
  }

  update(id: string, request: UpdateUtilisateur): Observable<Utilisateur> {
    return this.http.put<Utilisateur>(`${this.endpoint}/${id}`, request);
  }

  delete(id: string): Observable<void> {
    return this.http.delete<void>(`${this.endpoint}/${id}`);
  }
}
